Last-Update: 2025-12-24

--- kwin-6.3.6.orig/src/kcms/decoration/ui/main.qml
+++ kwin-6.3.6/src/kcms/decoration/ui/main.qml
@@ -51,6 +51,7 @@ KCM.AbstractKCM {
         },
         NewStuff.Action {
             configFile: "window-decorations.knsrc"
+            visible: false
             text: i18nc("@action:button as in, \"Get New Window Decorations\"", "Get New…")
 
             onEntryEvent: (entry, event) => {
--- kwin-6.3.6.orig/src/kcms/effects/ui/main.qml
+++ kwin-6.3.6/src/kcms/effects/ui/main.qml
@@ -23,7 +23,7 @@ ScrollViewKCM {
 
     actions: NewStuff.Action {
         text: i18nc("@action:button get new KWin effects", "Get New…")
-        visible: KAuthorized.authorize(KAuthorized.GHNS)
+        visible: false
         configFile: "kwineffect.knsrc"
         onEntryEvent: (entry, event) => {
             if (event === NewStuff.Engine.StatusChangedEvent) {
--- kwin-6.3.6.orig/src/kcms/scripts/ui/main.qml
+++ kwin-6.3.6/src/kcms/scripts/ui/main.qml
@@ -25,7 +25,7 @@ KCMUtils.ScrollViewKCM {
         },
         NewStuff.Action {
             text: i18nc("@action:button get new KWin scripts", "Get New…")
-            visible: KAuthorized.authorize(KAuthorized.GHNS)
+            visible: false
             configFile: "kwinscripts.knsrc"
             onEntryEvent: (entry, event) => {
                 if (event === NewStuff.Engine.StatusChangedEvent) {
--- kwin-6.3.6.orig/src/kcms/tabbox/main.cpp
+++ kwin-6.3.6/src/kcms/tabbox/main.cpp
@@ -72,12 +72,12 @@ KWinTabBoxConfig::KWinTabBoxConfig(QObje
     tabWidget->addTab(m_primaryTabBoxUi, i18n("Main"));
     tabWidget->addTab(m_alternativeTabBoxUi, i18n("Alternative"));
 
-    KNSWidgets::Button *ghnsButton = new KNSWidgets::Button(i18n("Get New Task Switcher Styles…"), QStringLiteral("kwinswitcher.knsrc"), widget());
-    connect(ghnsButton, &KNSWidgets::Button::dialogFinished, this, [this](auto changedEntries) {
-        if (!changedEntries.isEmpty()) {
-            initLayoutLists();
-        }
-    });
+//    KNSWidgets::Button *ghnsButton = new KNSWidgets::Button(i18n("Get New Task Switcher Styles…"), QStringLiteral("kwinswitcher.knsrc"), widget());
+//    connect(ghnsButton, &KNSWidgets::Button::dialogFinished, this, [this](auto changedEntries) {
+//        if (!changedEntries.isEmpty()) {
+//            initLayoutLists();
+//        }
+//    });
 
     QHBoxLayout *buttonBar = new QHBoxLayout();
     QStyle *style = widget()->style();
@@ -85,7 +85,7 @@ KWinTabBoxConfig::KWinTabBoxConfig(QObje
     buttonBar->setContentsMargins(style->pixelMetric(QStyle::PM_LayoutLeftMargin), 0, style->pixelMetric(QStyle::PM_LayoutRightMargin), style->pixelMetric(QStyle::PM_LayoutBottomMargin));
     QSpacerItem *buttonBarSpacer = new QSpacerItem(40, 20, QSizePolicy::Expanding, QSizePolicy::Minimum);
     buttonBar->addItem(buttonBarSpacer);
-    buttonBar->addWidget(ghnsButton);
+//    buttonBar->addWidget(ghnsButton);
 
     QVBoxLayout *layout = new QVBoxLayout(widget());
     layout->setContentsMargins(0, 0, 0, 0);
